assets/src/features/devices/BlurProcessor.tsx

```diff
@@ -0,0 +1,292 @@
+import { UseCameraResult } from "@jellyfish-dev/react-client-sdk";
+import { FilesetResolver, ImageSegmenter, ImageSegmenterCallback } from "@mediapipe/tasks-vision";
+import { useState, useRef, useReducer, useEffect, useCallback } from "react";
+import { useApi } from "../../jellyfish.types";
+import BlurWorker from "./BlurProcessorWorker?worker";
+
+export function useBlur<T>(video: UseCameraResult<T>): {
+  blur: boolean;
+  setBlur: (status: boolean) => void;
+  video: UseCameraResult<T>;
+} {
+  const [blur, setBlur] = useState(false);
+  const [prevStream, setPrevStream] = useState(video.stream);
+  const processor = useRef<BlurProcessor | null>(null);
+  const [, rerender] = useReducer((p) => p + 1, 0);
+  const api = useApi();
+  const apiRef = useRef(api);
+
+  useEffect(() => {
+    apiRef.current = api;
+  }, [api]);
+
+  useEffect(() => {
+    if (!video.stream) {
+      if (processor.current) {
+        processor.current.destroy();
+        processor.current = null;
+        setPrevStream(null);
+      }
+      return;
+    }
+
+    if (prevStream === video.stream) return;
+    setPrevStream(video.stream);
+    processor.current = new BlurProcessor(video.stream);
+
+    return () => {
+      processor.current?.destroy();
+      processor.current = null;
+      setPrevStream(null);
+    };
+  }, [video.stream]);
```

Comment:
I would add this simple `if(!blur) return;` to prevent the blur processor from starting if the checkbox is unchecked.

```suggestion
useEffect(() => {
    if (!video.stream) {
      if (processor.current) {
        processor.current.destroy();
        processor.current = null;
        setPrevStream(null);
      }
      return;
    }
    if(!blur) return;

    if (prevStream === video.stream) return;
    setPrevStream(video.stream);
    processor.current = new BlurProcessor(video.stream);

    return () => {
      processor.current?.destroy();
      processor.current = null;
      setPrevStream(null);
    };
  }, [video.stream, blur]);
```